src/main/java/uk/ac/ebi/atlas/experimentpage/ExpressionAtlasContentService.java

```diff
@@ -115,19 +174,62 @@ public ExpressionAtlasContentService(
                     (BaselineExperiment) experiment, contentType);
         } else if (experiment.getType().isRnaSeqBaseline()) {
-            return rnaSeqBaselineExperimentExternallyAvailableContentService.list(
+            List list = rnaSeqBaselineExperimentExternallyAvailableContentService.list(
                     (BaselineExperiment) experiment, contentType);
+            List list2 =  externalResourceType.equals("geo") ?
+                    rnaSeqBaselineExperimentExternallyAvailableContentServiceGeo.list(
+                    (BaselineExperiment) experiment, contentType) :
+                    externalResourceType.equals("ega") ?
+                    rnaSeqBaselineExperimentExternallyAvailableContentServiceEga.list(
+                            (BaselineExperiment) experiment, contentType) :
+                    rnaSeqBaselineExperimentExternallyAvailableContentServiceEna.list(
+                            (BaselineExperiment) experiment, contentType);
+            list.addAll(list2);
+            return list;
         } else if (experiment.getType().isRnaSeqDifferential()) {
-            return rnaSeqDifferentialExperimentExternallyAvailableContentService.list(
+            List list = rnaSeqDifferentialExperimentExternallyAvailableContentService.list(
                     (DifferentialExperiment) experiment, contentType);
+            List list2 =  externalResourceType.equals("geo") ?
+                    rnaSeqDifferentialExperimentExternallyAvailableContentServiceGeo.list(
+                            (DifferentialExperiment) experiment, contentType) :
+                    externalResourceType.equals("ega") ?
+                    rnaSeqDifferentialExperimentExternallyAvailableContentServiceEga.list(
+                            (DifferentialExperiment) experiment, contentType) :
+                    rnaSeqDifferentialExperimentExternallyAvailableContentServiceEna.list(
+                            (DifferentialExperiment) experiment, contentType);
+            list.addAll(list2);
+            return list;
         } else {
-            return microarrayExperimentExternallyAvailableContentService.list(
+            List list = microarrayExperimentExternallyAvailableContentService.list(
                     (MicroarrayExperiment) experiment, contentType);
+            List list2 =  externalResourceType.equals("geo") ?
+                    microarrayExperimentExternallyAvailableContentServiceGeo.list(
+                            (MicroarrayExperiment) experiment, contentType) :
+                    externalResourceType.equals("ega") ?
+                    microarrayExperimentExternallyAvailableContentServiceEga.list(
+                            (MicroarrayExperiment) experiment, contentType) :
+                    microarrayExperimentExternallyAvailableContentServiceEna.list(
+                            (MicroarrayExperiment) experiment, contentType);
+            list.addAll(list2);
+            return list;
```

Comment:
You can use `var` here. Also, give some descriptive names to your variables. `list` or `list2` don’t carry a lot of meaning to whomever is reading the code.

Also, in this case I think [a `switch` statement](https://docs.oracle.com/javase/7/docs/technotes/guides/language/strings-switch.html) will be more readable.

Reply:
I would agree with @alfonsomunozpomer review comments.